molecule_vagrant/modules/vagrant.py

```diff
@@ -508,79 +351,57 @@ def _get_config(self):
+            instances=self._get_vagrant_instances(),
         )
         molecule.util.write_file(self._vagrantfile, template)
 
-    def _write_vagrantfile_config(self, data):
-        molecule.util.write_file(
-            self._config.driver.vagrantfile_config, molecule.util.safe_dump(data)
-        )
-
-    def _write_configs(self):
-        self._write_vagrantfile_config(self._get_vagrant_config_dict())
-        self._write_vagrantfile()
-
     def _get_vagrant(self):
-        v = vagrant.Vagrant(
+        return vagrant.Vagrant(
             out_cm=self.stdout_cm,
             err_cm=self.stderr_cm,
             root=os.environ["MOLECULE_EPHEMERAL_DIRECTORY"],
         )
 
-        return v
-
-    def _get_vagrant_config_dict(self):
-        d = {
-            "config": {
-                # NOTE(retr0h): Options provided here will be passed to
-                # Vagrant as "config.#{key} = #{value}".
-                "options": {
-                    # NOTE(retr0h): `synced_folder` does not represent the
-                    # actual key used by Vagrant.  Is used as a flag to
-                    # simply enable/disable shared folder.
-                    "synced_folder": False,
-                    "ssh.insert_key": True,
-                }
-            },
-            "platform": {
-                "box": self._module.params["platform_box"],
-                "box_version": self._module.params["platform_box_version"],
-                "box_url": self._module.params["platform_box_url"],
-            },
-            "instance": {
-                "name": self._module.params["instance_name"],
-                "interfaces": self._module.params["instance_interfaces"],
-                "raw_config_args": self._module.params["instance_raw_config_args"],
-            },
-            "provider": {
-                "name": self._module.params["provider_name"],
-                # NOTE(retr0h): Options provided here will be passed to
-                # Vagrant as "$provider_name.#{key} = #{value}".
-                "options": {
-                    "memory": self._module.params["provider_memory"],
-                    "cpus": self._module.params["provider_cpus"],
-                },
-                "raw_config_args": self._module.params["provider_raw_config_args"],
-                "override_args": self._module.params["provider_override_args"],
-            },
-        }
-
-        d["config"]["options"].update(
-            molecule.util.merge_dicts(
-                d["config"]["options"], self._module.params["config_options"]
-            )
-        )
-
-        d["provider"]["options"].update(
-            molecule.util.merge_dicts(
-                d["provider"]["options"], self._module.params["provider_options"]
-            )
-        )
-
-        return d
+    def _get_vagrant_instances(self):
+        return [
+            {
+                "name": instance["name"],
+                "box": instance.get("box", "generic/alpine310"),
```

Comment:
hm. I'm not convinced that hard coding the default image in the python module is a good idea.
If the image gets removed:
- if it's located in the ``create.yml``, it's easy for a user to override, while waiting for us to fix.
- if it's in the python module, it's harder imho.

At least, I understand that it should prevent issues like bug #19

Reply:
The reason it's hardcoded in the module is because it can no longer be specified directly in `create.yml`, since it's part of the new `instances` variable instead. However, perhaps we should have an explicit check in the module for required fields in the `instances` variable? I think it makes sense to require at least `box` to be set by the user.